docs: update analyzing-bundles with Expo Atlas (#29649)
expo/expo: docs/pages/guides/analyzing-bundles.mdx

## Analyzing bundle size

> The libraries used in a project influence the size of the production JavaScript bundle. Starting from **Expo SDK 51**, you can use [Expo Atlas](https://github.com/expo/expo-atlas#readme) to visualize the production bundle and identify which libraries contribute to the bundle size.

Author: Good point, we could keep the source map analyzer below with a warning that it will eventually be deprecated (or at least add a warning that it's for SDK 50 and below)

> The libraries used in a project influence the size of the production JavaScript bundle. Starting from **Expo SDK 51**, you can use [Expo Atlas](https://github.com/expo/expo-atlas#readme) to visualize the production bundle and identify which libraries contribute to the bundle size.

Reviewer: We can also use a `Collapsible` for SDK 50 instructions and I agree about adding warning/deprecation callout.

> The libraries used in a project influence the size of the production JavaScript bundle. Starting from **Expo SDK 51**, you can use [Expo Atlas](https://github.com/expo/expo-atlas#readme) to visualize the production bundle and identify which libraries contribute to the bundle size.

Author: Done

> The libraries used in a project influence the size of the production JavaScript bundle. Starting from **Expo SDK 51**, you can use [Expo Atlas](https://github.com/expo/expo-atlas#readme) to visualize the production bundle and identify which libraries contribute to the bundle size.

Author: I think the `Collapsible` might be a bit much, since the old docs also contains step components etc. So I added the docs back, but used the titles:
- Analyzing bundle size with Atlas
- Analyzing bundle size with source-map-explorer

> The libraries used in a project influence the size of the production JavaScript bundle. Starting from **Expo SDK 51**, you can use [Expo Atlas](https://github.com/expo/expo-atlas#readme) to visualize the production bundle and identify which libraries contribute to the bundle size.

Reviewer: Thanks, Cedric!

> ### Using Atlas with `expo start`

Author: I was on the fence of this one. It's a bit verbose to have in the title itself, here we just reference the "start command" (as `expo start`). We also list out the full command in the terminal block here.

No strong opinions though, if you think we should do this, let's add it.

> ### Using Atlas with `expo start`

Reviewer: Yeah, I can understand. The reason I suggested using the `npx` prefix is because the `npx expo start` is like a universal command that we say everywhere. It becomes easier for the user to connect the dots as well as quickly review this doc. Also, another thing to note here is that `expo start` (even through that's not the intention) is somewhat relates to the older CLI iirc. We can avoid any room for confusion when onboarding a user persona who might be familiar with our tools updates but at the same might not be too up to date.

> You can use Expo Atlas with the local development server. This method allows Atlas to update whenever you change any code within your project. 

Reviewer: ```suggestion
You can use Expo Atlas with the local development server. This method allows Atlas to update whenever you change any code in your project. 
```

> Once your app is up and running on Android, iOS, and/or web, you can open Atlas through the [dev tools plugin menu](/debugging/devtools-plugins/#using-a-dev-tools-plugin) using <kbd>shift</kbd> + <kbd>m</kbd>.

Reviewer: ```suggestion
Once your app is running using the local development server on Android, iOS, and/or web, you can open Atlas through the [dev tools plugin menu](/debugging/devtools-plugins/#using-a-dev-tools-plugin) using <kbd>shift</kbd> + <kbd>m</kbd>.
```

> <Terminal cmd={[
>   '# Start the local development server with Atlas',
>   '$ EXPO_UNSTABLE_ATLAS=true npx expo start',
>   '',
>   '# Optionally, run the local development server in production mode',
>   '$ EXPO_UNSTABLE_ATLAS=true npx expo start --no-dev', 
> ]} />

Reviewer: ```suggestion
<Terminal cmd={[
  '# Start the local development server with Atlas',
  '$ EXPO_UNSTABLE_ATLAS=true npx expo start',
]} />
```

See below.

> > **Changing development mode to production**
> >
> > Expo starts in development mode by default. Development mode may turn off optimizations that are enabled in production mode. You can also start the local development server in production mode, which provides a more accurate representation of the production bundle size.

Reviewer: ```suggestion
#### Changing development mode to production

By default, Expo starts the local development server in [development mode](/workflow/development-mode/#development-mode). Development mode disables some optimizations that are enabled in [production mode](/workflow/development-mode/#production-mode). You can also start the local development server in production mode to get a more accurate representation of the production bundle size:

<Terminal cmd={[
  '# Run the local development server in production mode',
  '$ EXPO_UNSTABLE_ATLAS=true npx expo start --no-dev', 
]} />
```

Let's make production mode info part of the main content by including it as a sub section instead of using a callout and also split the command the since the context in the previous callout was below the callout (not a good reading experience for the user since we're telling them the command first without any context).

> ### Using Atlas with `expo export`

Reviewer: ```suggestion
### Using Atlas with `npx expo export`
```

Author: Same as: https://github.com/expo/expo/pull/29649#discussion_r1638785598

You can also use Expo Atlas when generating a production bundle for your app or EAS Update. Atlas generates a **.expo/atlas.jsonl** file during export, which you can share and open without having access to the project.

<Terminal cmd={[
  '# Export your app for all platforms',
  '$ EXPO_UNSTABLE_ATLAS=true npx expo export',
  '',
  '# Open the generated Atlas file',
  '$ npx expo-atlas .expo/atlas.jsonl',
]} />

> You can also specify the platforms you want to analyze, using the `--platform` flag. Atlas will gather the data for the exported platforms only.

Reviewer: ```suggestion
You can also specify the platforms you want to analyze using the `--platform` option. Atlas will gather the data for the exported platforms only.
```

In our CLI tools, we often call options instead of flag. Maybe for consistency we should use the same term here.

Also, does this `--platform` option/flag requires passing the platform name? If so, may be we should mention that or provide a list fo all platforms (as it is not obvious to someone using this tool initially).

Author: `--platform` requires a name, yes. But, if you omit it, we error and tell it needs a value:

<img width="280" alt="image" src="https://github.com/expo/expo/assets/1203991/7f07794e-7dcc-4c94-a273-9d236de92de0">

Also in the `--help` output, we put everything in there:

<img width="553" alt="image" src="https://github.com/expo/expo/assets/1203991/e67d6f12-1ef1-4f7a-94bb-a7d89138c124">

I do think these "path-finding errors" are good enough to get users unblocked. If people have use it once, they should know the option.

Author: Speaking of, could add the "allowed options" to the error message. I think that's a great addition. Right now, we only do that for bad input:

<img width="566" alt="image" src="https://github.com/expo/expo/assets/1203991/338fc984-7ef3-45e4-a59d-3ce8b1ffbf3e">
